src/kakarot/instructions/stop_and_math_operations.cairo

```diff
@@ -250,14 +251,25 @@ namespace StopAndMathOperations {
         jmp end;
 
         SIGNEXTEND:
+        // Examples:
+        // SignExtend 85
+        // x = 01010101, b = 1
+        // Extend x to be two bytes:
+        // x = 00000000 01010101
+        // ---
+        // SignExtend -28
+        // x = 11100100, b = 1
+        // Extend x to be two bytes:
+        // x = 11111111 10101010
+        let bitwise_ptr = cast([fp - 4], BitwiseBuiltin*);
         let range_check_ptr = [ap - 2];
         let popped = cast([ap - 1], Uint256*);
 
-        // TODO: see https://github.com/kkrt-labs/kakarot/issues/677
+        let value = uint256_signextend(popped[1], popped[0]);
```

Comment:
`let result`, this is the result, the following lines are more for the compile

Reply:
Modified